redis/redisLimiter.go

```diff
@@ -239,6 +239,18 @@ func (rl *rateLimiter) Rate(mode Mode) int {
 	return int(rl.freezeFailureConfig.maxFailures)
 }
 
+// ExtendSecurityMode extends the security mode to the maximum limit
+func (rl *rateLimiter) ExtendSecurityMode(key string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	_, err := rl.storer.SetExpire(ctx, key, rl.securityModeFailureConfig.limitPeriod)
```

Comment:
i think we better use here `ExpireXX`, which will set expiry only when the key has an existing expiry

Reply:
this is not implemented yet in our redis wrapper, but can be easily added

Reply:
pushed